topdown/http.go

```diff
@@ -48,16 +49,30 @@ var requiredKeys = ast.NewSet(ast.StringTerm("method"), ast.StringTerm("url"))
 
 var client *http.Client
 
+// httpSendBuiltinCacheKey is the key in the builtin context cache that
+// points to the http.send() specific cache resides at.
+const httpSendBuiltinCacheKey = "HTTP_SEND_CACHE_KEY"
+
+var httpSendCacheSync sync.RWMutex
```

Comment:
I'm not sure I follow the locking logic. If we assume query eval is single threaded then this locking is not needed (since the built-in context isn't accessed concurrently). For multi-threaded query eval we would probably want to start by ensuring that concurrent calls return the same answer, however, as it's currently implemented it would be possible to see multiple results inside the policy since we don't hold a write lock for the entire call or throw away all-but-one of the responses in the speculative case.

For simplicity, I'd probably just remove the locking unless it's required for now.

Reply:
Thats a good point, I'll just remove it for now. I made the assumption it was ok to race for the first entry in the cache but yea, if the responses were different it would potentially cause problems even if its still 1 thread per eval and just a shared cache between two different evals.